src/compiler/checker.ts

```diff
@@ -20310,7 +20310,8 @@ namespace ts {
                     shouldSkipElaboration = true; // Retain top-level error for interface implementing issues, otherwise omit it
                 }
                 if (props.length === 1) {
-                    const propName = symbolToString(unmatchedProperty);
+                    const nameType = getSymbolLinks(unmatchedProperty).nameType;
+                    const propName = nameType && nameType.flags & TypeFlags.EnumLiteral ? `[${symbolToString(nameType.symbol, nameType.symbol.valueDeclaration)}]` : symbolToString(unmatchedProperty);

```

Comment:
We can pass only one argument and then can destructure in symbolToString function right? 

```suggestion
                    const propName = nameType && nameType.flags & TypeFlags.EnumLiteral ? `[${symbolToString(nameType.symbol)}]` : symbolToString(unmatchedProperty);

```